substrate/client/authority-discovery/src/worker.rs

```diff
@@ -906,7 +906,7 @@ where
 		if new_record.creation_time > current_record_info.creation_time {
 			let peers_that_need_updating = current_record_info.peers_with_record.clone();
 			self.network.put_record_to(
-				new_record.record.clone(),
+				new_record.record.clone().into(),
```

Comment:
`network.put_record_to()` should be changed to accept generic `Record` from `sc_network_types::kad` and this `.into()` should not be needed.